### ISSUE TITLE
Game freezing on startup as of version 1.20.x

### ISSUE THREAD

**Issue by jrugia** [Novum/vkQuake#512]
As of release 1.20.0 up to 1.20.2 game is freezing upon startup. Reverting to 1.13.1 works fine.

Between 1.13.1 and the new release, the only thing that I see different in terminal are these 3 lines:

Unknown command "gl_subdivide_size"
Unknown command "gl_triplebuffer"
Unknown command "vid_bpp"

![1](https://user-images.githubusercontent.com/13750621/177035456-cd5a2b12-37d9-40fc-ad90-01197c32f64c.jpg)

**Comment by Novum** [comment]
Thanks for the report. The three warnings are harmless

Not sure how I can fix this, doesn't happen on my AMD Laptop. What CPU is this?

**Comment by Novum** [comment]
Could you also attach your `config.cfg` and `autoexec.cfg` (if it exists)?

**Comment by Novum** [comment]
Could you try latest? e6f8aaf6882cb0650cd7363cddb8020984945513

**Comment by jrugia** [comment]
> Also try latest [e6f8aaf](https://github.com/Novum/vkQuake/commit/e6f8aaf6882cb0650cd7363cddb8020984945513)

Still same issue unfortunately.



> Could you also attach your `config.cfg` and `autoexec.cfg` (if it exists)?

I had config.cfg but deleted it before attempting to run each new build.

**Comment by Novum** [comment]
Just makes little sense, I have not seen this in a lot of runs on very similar hardware. At this point I would need to get a callstack/dump to reason about this.

**Comment by jrugia** [comment]
How can i help?  I can try to obtain debug data but will need guidance how to do so.

**Comment by Novum** [comment]
`gdb ./vkquake`
then give commands `interrupt process`, `thread apply all bt`

**Comment by jrugia** [comment]
> `gdb ./vkquake` then after it hangs give commands `interrupt process`, `thread apply all bt`
> 
> You might also need to build with debug information for this to be useful: `rm -rf build && meson build --buildtype=debugoptimized && ninja -C build`

Got the debug build compiled. Attempted to run it through the debugger doesn't produce anything.

Attempting to run the game produces the following errors in terminal:  [terminal.log](https://github.com/Novum/vkQuake/files/9035353/terminal.log)

The debug build freezes with `taskset -c 0` whereas the release build does not.

**Comment by Novum** [comment]
You need to `Ctrl+C` (in the console) to break into the debugger. Sorry was unclear with my instructions.

**Comment by sezero** [comment]
Probably amounts to nothing, but you seem to be running without vkquake.pak: Does it change anything if you do?

**Comment by jrugia** [comment]
> You need to `Ctrl+C` (in the console) to break into the debugger. Sorry was unclear with my instructions.

Still nothing, it basically stays like this, even just running `bt` says "No Stack":

![Screenshot at 2022-07-03 21-13-03](https://user-images.githubusercontent.com/13750621/177054034-01ddecc7-8e9c-4582-9f0a-bade001ecc7c.png)


> Probably amounts to nothing, but you seem to be running without vkquake.pak: Does it change anything if you do?

vkquake.pak is present in the "id1" folder (I did remove the old one and replaced it with the one from 1.20.2 build)

**Comment by sezero** [comment]
> Still nothing, it basically stays like this, even just running `bt` says "No Stack":

You didn't "run" it: Enter `r` to run the executable.

> vkquake.pak is present in the "id1" folder (I did remove the old one and replaced it with the one from 1.20.2 build)

The screenshot you gave us doesn't have the custom console background which is present in vkquake.pak.

**Comment by jrugia** [comment]
> > Still nothing, it basically stays like this, even just running `bt` says "No Stack":
> 
> You didn't "run" it: Enter `r` to run the executable.
> 

Thanks, got the log output now:  [gdb.log](https://github.com/Novum/vkQuake/files/9035385/gdb.log)


> > vkquake.pak is present in the "id1" folder (I did remove the old one and replaced it with the one from 1.20.2 build)
> 
> The screenshot you gave us doesn't have the custom console background which is present in vkquake.pak.

That's weird, I always use the vkquake.pak generated at compile time and place it in the id1 folder.

**Comment by Novum** [comment]
It hangs in `Task_AddDependency` - that's a new one I haven't seen yet.

**Comment by Novum** [comment]
It's hanging on a mutex, which makes no sense. Nothing else seems to be holding that mutex.

**Comment by Novum** [comment]
What Linux distribution is that btw? I wonder if there is some bug in glibc/SDL2 that has been fixed, although that seems unlikely. The compiler is quite old though.

**Comment by jrugia** [comment]
> What Linux distribution is that btw? I wonder if there is some bug in glibc/SDL2 that has been fixed, although that seems unlikely. The compiler is quite old though.

It's Debian 9. I'm aware it's quite old but I've modified it to support a lot of newer features over the years.  I had already tried to compile with a newer gcc which produced same issue.

Since you mentioned SDL2 I tried to compile against v2.0.22 (latest) and sure enough this has solved my issue. I've successfully launched the game multiple times now (both debug and release builds) and there are no more mutex errors.


Thank you for your help.
